libs/s3/s3_versioning_common_test_lib.py

```diff
@@ -272,14 +271,12 @@ def check_get_head_object_version(s3_ver_test_obj: S3VersioningTestLib,
                                                                 key=object_name,
                                                                 version_id=version_id)
         else:
-            head_response = s3_ver_test_obj.object_info(bucket_name=bucket_name,
-                                                        key=object_name)
+            head_response = s3_test_obj.object_info(bucket_name=bucket_name, key=object_name)
         assert_utils.assert_true(head_response[0], head_response[1])
         if version_id:
-            assert_utils.assert_equal(
-                head_response[1]["ResponseMetadata"]["VersionId"], version_id)
+            assert_utils.assert_equal(head_response[1]["VersionId"], version_id)
```

Comment:
instead of assert CTException may be re raised mentioning response in message

Reply:
We are handling error message assertions in case the test is expecting an error from GET and HEAD Object, which is why we have except blocks in the code.